animations/concepts.md

```diff
@@ -0,0 +1,298 @@
+---
+layout: page
+title: "Flutter animation concepts"
+permalink: /animations/concepts.html
+---
+
+* TOC Placeholder
+{:toc}
+
+
+This section describes the basic concepts behind the Flutter animation architecture.
+
+If you use Flutter's built-in implicit and transition widgets, you may not need to interact with all of the Flutter core animation concepts directly because most of the work required to code the animation is already done. But for more complex explicit animations, it's important to understand the animation concepts used to fine-tune and control your animations.
+
+## How it works
+Animation is the movement of an object or a change over time—like a fade in or out, or a bounce. An animation represents a value of a specific type that can change over time—specifically over the lifetime of the animation. Most animation widgets receive an `Animation` object as a parameter, from which the animation widget reads the current value of the animation and listens for changes to that value. It listens for "state" changes.
+
+Whenever the animation’s object value changes, the animation object notifies all the listeners (that were added with `addListener`). Typically, a `State` object that listens to an animation will call `setState` in its listener callback to notify the widget system that it needs to rebuild with the new value of the animation.
```

Comment:
1. I think a State object ALWAYS calls setState, doesn't it? Remove "Typically,"
2. will call => calls
3. This is a super important point.